packages/@n8n/decorators/src/module.ts

```diff
@@ -33,10 +50,89 @@ export class ModuleRegistry {
 	}
 
 	registerLifecycleHooks(hooks: ExecutionLifecycleHooks) {
-		for (const ModuleClass of registry.keys()) {
-			const instance = Container.get(ModuleClass);
-			if (instance.registerLifecycleHooks) {
-				instance.registerLifecycleHooks(hooks);
+		const handlers = Container.get(LifecycleMetadata).getHandlers();
+
+		for (const { handlerClass, methodName, eventName } of handlers) {
+			const instance = Container.get(handlerClass);
+
+			switch (eventName) {
+				case 'workflowExecuteAfter':
+					hooks.addHandler(
+						eventName,
+						async function (
+							this: { workflowData: IWorkflowBase },
+							runData: IRun,
+							newStaticData: IDataObject,
+						) {
+							const context: LifecycleContext = {
```

Comment:
maybe we should update the existing life-cycles code to use these context types as well, instead of having to restructure these object.